pkg/transport/generic/cmd/kubeconfig-resolver.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	"github.com/kubestellar/kubestellar/pkg/ctrlutil"
+)
+
+// resolveWDSKubeconfig resolves WDS kubeconfig using priority order
+// Priority 1: WdsClientOptions (handles --wds-kubeconfig flag)
+// Priority 2: Cluster-based discovery using WDS name (--wds-name)
+func resolveWDSKubeconfig(options *TransportOptions, logger klog.Logger) (*rest.Config, string, error) {
+	// Priority 1: Try WdsClientOptions first (this handles --wds-kubeconfig)
+	if wdsConfig, err := options.WdsClientOptions.ToRESTConfig(); err == nil {
```

Comment:
Did you know that when run by a Pod in the KubeFlex hosting cluster, with `options.WdsClientOptions` having the empty string for the `loadingRules.ExplicitPath`, `ToRESTConfig()` will succeed and return a client config for accessing the KubeFlex hosting cluster?